google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -808,14 +807,114 @@ void RenameObject(google::cloud::storage::Client client, int& argc,
   (std::move(client), bucket_name, old_object_name, new_object_name);
 }
 
+void SetObjectEventBasedHold(google::cloud::storage::Client client, int& argc,
+                             char* argv[]) {
+  if (argc != 3) {
+    throw Usage{"set-object-event-based-hold <bucket-name> <object-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto object_name = ConsumeArg(argc, argv);
+  //! [set event based hold] [START storage_set_event_based_hold]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string object_name) {
+    gcs::ObjectMetadata original =
+        client.GetObjectMetadata(bucket_name, object_name);
+    gcs::ObjectMetadata metadata = client.PatchObject(
+        bucket_name, object_name,
+        gcs::ObjectMetadataPatchBuilder().SetEventBasedHold(true),
+        gcs::IfMetagenerationMatch(original.metageneration()));
+    std::cout << "The event hold for object " << metadata.name()
+              << " in bucket " << metadata.bucket() << " is "
+              << (metadata.event_based_hold() ? "enabled" : "disabled")
```

Comment:
This metadata field will have a similar issue with `Fields("name")` not selecting `eventBasedHold` and the library defaults to `false`.

Reply:
True.  That is also true of most of the other fields, we discussed this in #934 (but did not record as an ADR, just sent #1358 to fix that) and decided to keep most simple fields not wrapped in `optional<>`.  Are you requesting a change or just observing?

Reply:
If this was previously discussed and will now be documented, I think that's fair. I find it slightly confusing if I were to use `fields` to select a field and find boolean and integers fields with `false` and `0` respectively.

It would be helpful to use `optional<>` for these fields when selecting `fields`, but I also understand the ease-of-use when using primitives over using `optional<>` which may not be exercised by all C++ developers _yet_.

Keep as-is.